Technical-Steering-Committee/guidelines/breaking-changes.md

```diff
@@ -0,0 +1,136 @@
+
+    - A deprecation notice appears in the documentation in the API reference documentation and places where the functionality is described for the active and under development releases.
+    - In the case of programmatic Java or Node.js APIs, deprecation mechanisms provided in the programming language are used (e.g. `@Deprecated` annotation)
+    - Release notes mention the deprecation.
+    - There are no functional changes.
+    - By default, there will be no warnings emitted for such deprecations at runtime.
+
+2. Runtime Deprecation
+
+    - Emits a warning at runtime on first use of the deprecated API or when CLI is used
+    - Can be disabled (environment variable `ZOWE_DEPRECATION_WARN=0` for CLI and  configuration option `zowe.setup.deprecation.warning` in `zowe.yaml` for z/OS Components).
+    - Can be changed to cause an runtime error (by `ZOWE_DEPRECATION_RUNTIME_ERROR=1` and configuration option `zowe.setup.deprecation.runtime-error` in `zowe.yaml` for z/OS Components).
+
+3. End-of-Life
+
+    - Backward-incompatible changes including complete removal of such APIs or functionality can occur at any time.
+
+Apply the `notable change` label to all pull requests that introduce Documentation-Only Deprecations. Such deprecations have no impact on code execution. Thus, they are not breaking changes.
+
+Documentation-Only Deprecations can land in a minor release. They can change to a Runtime Deprecation after six months. They can not change to a End-of-Life until the next major release and need to be in End-of-Life at least for another six months.
+
+No API can change to End-of-Life without going through a Runtime Deprecation cycle. There is no rule that deprecated code must progress to End-of-Life. Documentation-Only and Runtime Deprecations can remain in place for an unlimited duration.
+
+Communicate pending deprecations and associated mitigations with the ecosystem as soon as possible. If possible, do it before the pull request adding the deprecation lands on the master branch.
+
+### Example
+
+1. A planned breaking change is discussed at Zowe architecture call and Zowe playback demo event.
+
+2. The breaking change is brought to TSC and approved.
+
+3. The documentation of active and development releases is updated and the change is now Documentation-Only Deprecation. If there is a new functionality that replaces the deprecated one, it needs to be made available on active and development branches at least at this moment.
+
+4. After at least six months and no push back of the community, the planned change moves to  Runtime Deprecation.
+
+5. The API or functionality can be removed at the new major release if that happens at least after six months since Runtime Deprecation and no concerns from the community. This requires another TSC review.
+
+## Squad Responsibilities
+
+Each squad documents for the components owned by the squads in [Zowe Docs](https://docs.zowe.org/):
+
+ - What is public and internal API
```

Comment:
You mean definition of how the squad thinks about what is public and internal?

Reply:
Yes. I tried to reword it better.